go/apps/api/routes/v2_keys_create_key/handler.go

```diff
@@ -233,12 +233,31 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
 					Meta:        []byte("{}"),
 				})
 				if err != nil {
-					return fault.Wrap(err,
-						fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
-						fault.Internal("failed to create identity"),
-						fault.Public("Failed to create identity."),
-					)
+					// Incase of duplicate key error just find existing identity
+					if !db.IsDuplicateKeyError(err) {
+						return fault.Wrap(err,
+							fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
+							fault.Internal("failed to create identity"),
+							fault.Public("Failed to create identity."),
+						)
+					}
+
+					identity, err = db.Query.FindIdentity(ctx, h.DB.RO(), db.FindIdentityParams{
```

Comment:
technically there could be replication lag, where the RO replica doesn't see the write yet 
in reality that's probably not a huge concern though

then again, using a RW replica in this case is also fine cause it's pretty rare